python-backend/app/api/variances/resources/variance_resource.py

```diff
@@ -43,24 +60,70 @@ def get(self):
         }
 
+    @classmethod
+    def _build_filter(cls, model, field, op, argfield):
+        return {
+            'model': model,
+            'field': field,
+            'op': op,
+            'value': argfield
+        }
+
+    def _apply_filters_and_pagination(self, args):
 
-    def _apply_filters_and_pagination(self,
-                                      page_number=PAGE_DEFAULT,
-                                      page_size=PER_PAGE_DEFAULT,
-                                      application_status=None):
         status_filter_values = list(map(
             lambda x: x.variance_application_status_code,
             VarianceApplicationStatusCode.active()))
 
-        if application_status is not None:
-            status_filter_values = application_status.split(',')
+        conditions = []
+        if args["application_status"] is not None:
+            status_filter_values = args["application_status"].split(',')
+            conditions.append(self._build_filter('Variance' , 'variance_application_status_code', 'in',  status_filter_values))
+
+        if args["compliance_codes"] is not None:
+            compliance_codes_values = args["compliance_codes"].split(',')
+            conditions.append(
+                self._build_filter('Variance', 'compliance_article_id', 'in', compliance_codes_values))
+
+        if args["expiry_date_before"] is not None:
+            conditions.append(
+                self._build_filter('Variance', 'expiry_date', '<=', args["expiry_date_before"]))
+
+        if args["expiry_date_after"] is not None:
+            conditions.append(
+                self._build_filter('Variance', 'expiry_date', '>=', args["expiry_date_after"]))
+
+        if args["issue_date_before"] is not None:
+            conditions.append(
+                self._build_filter('Variance', 'issue_date', '<=', args["issue_date_before"]))
+
+        if args["issue_date_after"] is not None:
+            conditions.append(
+                self._build_filter('Variance', 'issue_date', '>=', args["issue_date_after"]))
+
+        if args["major"] is not None:
+            conditions.append(
+                self._build_filter('Mine', 'major_mine_ind', '==', args["major"]))
+
+        if args["search_terms"] is not None:
+            search_term_list = args["search_terms"].split(' ')
+            search_conditions = []
+            for search_term in search_term_list:
+                search_conditions.append(
+                    self._build_filter('Mine', 'mine_name', 'ilike', '%{}%'.format(search_term)))
+
+                search_conditions.append(
+                    self._build_filter('Mine', 'mine_no', 'ilike', '%{}%'.format(search_term)))
+
+            conditions.append({'or': search_conditions})
+
+        if args["region"] is not None:
+            region_list = args["region"].split(',')
+            conditions.append(self._build_filter('Mine', 'mine_region', 'in', region_list))
+
+        query = Variance.query.join(Mine)
```

Comment:
You can probably merge this and next line together, doesn't seem like query object is complicated enough to be its own thing